scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/BaseOxfordNanoporeProcessor.java

```diff
@@ -63,27 +68,65 @@ public PathType getPathType() {
       Files.walkFileTree(
           root.toPath(),
           new FileVisitor<Path>() {
+
+            /**
+             * Actions to take before visiting each individual file within the directory, mostly
+             * dealing with directory path.
+             */
             @Override
             public FileVisitResult preVisitDirectory(
                 Path path, BasicFileAttributes basicFileAttributes) throws IOException {
-              // Immediately check if we're dealing with the old format, move on if so.
-              if (POSITION.matcher(path.toString()).find()) {
-                log.debug("Skipping " + path + " because we found a Sequencer Position in it.");
+              log.debug("Pre-visit: " + path);
+
+              /**
+               * If directory matches criteria we know exclude the directory from consideration,
+               * don't go into the directory (SKIP_SUBTREE)
+               */
+              if (excludedDirectoryFormat(path) || olderThanCutoff(path)) {
+                log.debug("Skipping " + path + " because we found an excluded directory in it.");
                 return FileVisitResult.SKIP_SUBTREE;
               }
+
+              /**
+               * If directory matches criteria we know include the directory, add without looking
+               * inside directory (add, SKIP_SUBTREE). Checking if result of readsDirectoryForRun
+               * 'isDirectory' is an existence check.
+               */
               if (readsDirectoryForRun(path).anyMatch(Files::isDirectory)) {
                 log.debug("Adding " + path);
                 runDirectories.add(path.toFile());
                 return FileVisitResult.SKIP_SUBTREE;
               }
-              return FileVisitResult.CONTINUE;
+
+              // If directory has directories or fast5s in, keep going through directory (CONTINUE)
+              if (Arrays.asList(path.toFile().listFiles())
```

Comment:
If you've got an array, you can do either `Stream.of` or `Arrays.stream` to skip the list.

Reply:
Done